compiler/IREmitter/Payload/patches/object.c

```diff
@@ -126,3 +126,7 @@ VALUE (*sorbet_vm_Kernel_instance_variable_get_func(void))(VALUE obj, VALUE iv)
 VALUE (*sorbet_vm_Kernel_instance_variable_set_func(void))(VALUE obj, VALUE iv, VALUE value) {
     return rb_obj_ivar_set;
 }
+
+void sorbet_rbasic_set_class(VALUE obj, VALUE klass) {
+    RBASIC_SET_CLASS(obj, klass);
+}
```

Comment:
Do we need this? It looks like it's defined in `internal.h`, which we do include in the codegen-payload. If we need it because it's defined in terms of something that we don't have access to, could you add a comment noting that?

Reply:
(:portal-from: Slack DMs...)

Originally I did this because `RBASIC_SET_CLASS` was expanding to something that includes `typeof`, which makes clang angry because that's a gcc-ism. We worked out that setting `-std=gnu99` (instead of `c99`) fixes this. Going to PR that separately.